src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -2317,10 +2317,13 @@ void GenericThreadStackManagerImpl_OpenThread<ImplClass>::OnDnsResolveResult(otE
 
     error = FromOtDnsResponseToMdnsData(serviceInfo, type, dnsResult->mMdnsService, dnsResult->mServiceTxtEntry);
 
+    VerifyOrExit(error == CHIP_NO_ERROR, );
+
+    DeviceLayer::PlatformMgr().ScheduleWork(DispatchResolve, reinterpret_cast<intptr_t>(dnsResult));
```

Comment:
No, this is going to leak the resolution proxy over in Discovery_ImplPlatform that is waiting for us to respond....

I think the right fix for the problem this is trying to fix is in `HandleNodeIdResolve` over in `src/lib/dnssd/Discovery_ImplPlatform.cpp` to return in the various error conditions where it does `proxy->Release()` instead of continuin?  Then it would do, up front:
```
    if (CHIP_NO_ERROR != error)
    {
        proxy->OnNodeIdResolutionFailed(PeerId(), error);
        proxy->Release();
        return; /* this line is new */
    }
```
and never touch the `DnssdService` in the error cases.

Reply:
Thank you, I was trying to find the proper way to exit. So far I haven't had luck removing the hard fault by modifying Discovery_ImplPlatform.cpp, but I will give this a shot.

Reply:
This seems to work. Thanks for your help!